lib/src/sdk_env.dart

```diff
@@ -445,23 +461,13 @@ class ToolEnvironment {
     }
   }
 
-  /// Removes the dev_dependencies from the pubspec.yaml
-  /// Returns the backup file with the original content.
-  Future<File> _stripPubspecYaml(String packageDir) async {
-    final now = DateTime.now();
-    final backup = File(
-        p.join(packageDir, 'pana-${now.millisecondsSinceEpoch}-pubspec.yaml'));
-
-    final pubspec = File(p.join(packageDir, 'pubspec.yaml'));
-    final original = await pubspec.readAsString();
-    final parsed = yamlToJson(original);
-    parsed.remove('dev_dependencies');
-    parsed.remove('dependency_overrides');
-
-    await pubspec.rename(backup.path);
-    await pubspec.writeAsString(json.encode(parsed));
-
-    return backup;
+  /// Returns a (shallow) copy of the underlying JSON map from [pubspec] with
+  /// dev_dependencies and dependency_overrides removed.
+  Map<dynamic, dynamic> _stripPubspecYaml(Pubspec pubspec) {
+    final json = Map.from(pubspec.toJson());
+    json.remove('dev_dependencies');
+    json.remove('dependency_overrides');
+    return json;
   }
```

Comment:
Couldn't we also create a `pubspec.yaml` on the form:

`{"name": "dummy pana", "dependencies": {"$package": {"path": "/path/to/folder"}}`

Then use this to create the `.dart_tool/package_config.json` file, which I suppose is all we're doing here?
Information about outdated packages we can get from `pub outdated --json` which has `--no-dependency-overrides` and `--no-dev-dependencies`.

End result is probably the same, I'm not sure creating an empty pubspec is cleaner..

Reply:
Yes! That's even better!

Reply:
Though - I'd like to do the `pub outdated` later in another PR.

Reply:
If I was unclear I just meant this as a suggestion... Feel free to ignore or leave for later..

I'm not sure it's cleaner to make a dummy pubspec.. it's just more similar to what we do in global activate..

Reply:
I'll leave it for now